### PR TITLE
Projections from Prefix Values

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

This PR adds the capability of projection values out of the prefix fields of the zkd index.

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBZkdIndex.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ class RocksDBZkdIndexBase : public RocksDBIndex {
   std::vector<std::vector<basics::AttributeName>> const& coveredFields()
       const override {
     // The index only covers stored values
-    return _storedValues;
+    return _coveredFields;
```
Comment (markuspf) [inline arangod/RocksDBEngine/RocksDBZkdIndex.h:47]: The comment above is now out of date, I presume

File: arangod/RocksDBEngine/RocksDBZkdIndex.cpp [hunk 2]
```diff
@@ -219,12 +219,24 @@ class RocksDBZkdIndexIterator final : public IndexIterator {
   bool nextCoveringImpl(CoveringCallback const& callback,
                         uint64_t limit) override {
     struct CoveringData : IndexIteratorCoveringData {
-      explicit CoveringData(velocypack::Slice data) : _data(data) {}
-      VPackSlice at(size_t i) const override { return _data.at(i); }
+      CoveringData(VPackSlice prefixValues, VPackSlice storedValues)
```
Comment (markuspf) [inline arangod/RocksDBEngine/RocksDBZkdIndex.cpp:222]: is there a reason that this constructor is not marked explicit anymore?

**Review by markuspf — APPROVED** [review]
LGTM, modulo a question and a comment.

File: arangod/RocksDBEngine/RocksDBZkdIndex.cpp (hunk 2 above)
Comment (maierlars) [reply arangod/RocksDBEngine/RocksDBZkdIndex.cpp:222]: yes, it has now two parameters. They can't be used in implicit conversions, right?

**Pull request merged**